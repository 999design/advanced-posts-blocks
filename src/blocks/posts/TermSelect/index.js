/**
 * Internal dependencies
 */
import { buildTermsTree } from '../../../util/terms';
import { TreeSelect } from '@wordpress/components';

export default function TermSelect( { label, noOptionLabel, termList, selectedTermId, onChange, multiple } ) {
	const termsTree = buildTermsTree( termList );
	return (
		<TreeSelect
			{ ...{ label, noOptionLabel, onChange, multiple } }
			tree={ termsTree }
			selectedId={ selectedTermId }
		/>
	);
}
