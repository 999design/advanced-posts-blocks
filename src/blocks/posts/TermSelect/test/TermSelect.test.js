/**
 * External dependencies
 */
import renderer from 'react-test-renderer';
/**
 * Internal dependencies
 */
import TermSelect from '../index';

test( 'TermSelect Multiple', () => {
	const termList = [
		{ id: 2232, parent: 0, name: 'term1', slug: 'term1' },
		{ id: 2245, parent: 2232, name: 'term2', slug: 'term2' },
		{ id: 2249, parent: 0, name: 'term3', slug: 'term3' },
		{ id: 2246, parent: 2232, name: 'term4', slug: 'term4' },
	];
	const component = renderer.create(
		<TermSelect
			noOptionLabel={ 'All' }
			multiple={ true }
			termList={ termList }
			label={ 'Terms' }
			selectedTermId={ [ 2232 ] }
		/>,
	);
	const tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
