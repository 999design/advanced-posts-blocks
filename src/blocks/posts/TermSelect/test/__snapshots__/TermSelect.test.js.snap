// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermSelect Multiple 1`] = `
<div
  className="components-base-control"
>
  <div
    className="components-base-control__field"
  >
    <label
      className="components-base-control__label"
      htmlFor="inspector-select-control-0"
    >
      Terms
    </label>
    <select
      className="components-select-control__input"
      id="inspector-select-control-0"
      multiple={true}
      onChange={[Function]}
      value={
        Array [
          2232,
        ]
      }
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value={2232}
      >
        term1
      </option>
      <option
        value={2245}
      >
           term2
      </option>
      <option
        value={2246}
      >
           term4
      </option>
      <option
        value={2249}
      >
        term3
      </option>
    </select>
  </div>
</div>
`;
